File: src/services/visita.service.ts
/**
 * Servicio de visitas.
 * Registrar apertura y cierre de visitas.
 */
import apiClient from './api-client';
import { Visita } from '../types';

export interface AbrirVisitaRequest {
  farmacia_id: number;
  latitud?: number;
  longitud?: number;
}

export interface CerrarVisitaRequest {
  visita_id: number;
}

export const visitaService = {
  /**
   * Registrar apertura de visita con fecha/hora y GPS opcional.
   */
  async abrir(data: AbrirVisitaRequest): Promise<Visita> {
    const response = await apiClient.post<{ data: Visita }>('/visitas/abrir', data);
    return response.data.data;
  },

  /**
   * Registrar cierre de visita con fecha/hora de finalización.
   */
  async cerrar(data: CerrarVisitaRequest): Promise<void> {
    await apiClient.post('/visitas/cerrar', data);
  },
};

File: lib/services/visita_service.dart
import '../core/api_client.dart';

/// Servicio de visitas.
/// "Abrir visita" = crear/iniciar la ficha de visita.
/// "Cerrar visita" = cerrar actividad.
class VisitaService {
  /// Registrar inicio de visita (crear ficha si no existe).
  static Future<int> abrir(int peticionId) async {
    final response = await apiClient.post(
      '/visitas/abrir',
      data: {'peticion_id': peticionId},
    );
    return response.data['data']['ficha_id'] as int;
  }

  /// Registrar cierre de visita (cerrar actividad).
  static Future<void> cerrar(int peticionId) async {
    await apiClient.post(
      '/visitas/cerrar',
      data: {'peticion_id': peticionId},
    );
  }
}
